File: oa-portal/problems/a6-grid-1/starters/main.cpp
#include <bits/stdc++.h>
using namespace std;

const long long MOD = 1000000007LL;

long long countPaths(const vector<string>& grid) {
    // write your code here

    return 0;
}

int main() {
    int H, W;
    if (scanf("%d %d", &H, &W) != 2) return 0;
    vector<string> grid(H);
    for (int r = 0; r < H; r++) {
        static char buf[1005];
        scanf("%s", buf);
        grid[r] = buf;
    }

    printf("%lld\n", countPaths(grid));
    return 0;
}

File: oa-portal/problems/a6-grid-1/starters/main.java
import java.io.*;
import java.util.*;

public class Main {

    static final long MOD = 1000000007L;

    static long countPaths(String[] grid) {
        // write your code here

        return 0;
    }

    public static void main(String[] args) throws IOException {
        FastReader in = new FastReader();
        int H = in.nextInt(), W = in.nextInt();
        String[] grid = new String[H];
        for (int r = 0; r < H; r++) grid[r] = in.next();

        System.out.println(countPaths(grid));
    }

    static class FastReader {
        private final DataInputStream in = new DataInputStream(new BufferedInputStream(System.in, 1 << 16));
        int nextInt() throws IOException { return (int) nextLong(); }
        long nextLong() throws IOException {
            int b = in.read();
            while (b != '-' && (b < '0' || b > '9')) b = in.read();
            boolean neg = b == '-';
            if (neg) b = in.read();
            long v = 0;
            while (b >= '0' && b <= '9') { v = v * 10 + (b - '0'); b = in.read(); }
            return neg ? -v : v;
        }
        String next() throws IOException {
            int b = in.read();
            while (b <= ' ') b = in.read();
            StringBuilder sb = new StringBuilder();
            while (b > ' ') { sb.append((char) b); b = in.read(); }
            return sb.toString();
        }
    }
}
